src/browser/provider/built-in/dedicated/firefox/marionette-client/index.js

```diff
@@ -170,8 +176,8 @@ module.exports = class MarionetteClient {
         });
     }
 
-    async getScreenshotData () {
-        const frameData = await this._getScreenshotRawData();
+    async getScreenshotData (_, fullPage) {
```

Comment:
Why does the first parameter is necessary?

Reply:
this is a method of the `browserClient` interface.
The first parameter is required for Chrome.
I do not want to change parameters order, since the first one is more important.
Also I do not want to extend options from the first parameter with the `fullPage` property, since the first parameter is called `runtimeInfo`, so it should not know anything about the `fullPage` option

Reply:
Ok

Reply:
Currently, there is no common `BrowserClient` interface. There is a bunch of separate functions for Chrome and this class for Firefox. The code that calls browser-specific functions is quite different for different providers. 

I suggest removing the unused argument here and create a ticket to refactor browser clients to provide a common interface.

Reply:
Well. It's not a real interface of course, but still it is an interface, so I can not just remove this argument
https://github.com/DevExpress/testcafe/blob/9c96672c9c6a7fb50658a4c57f1e17ce645cd3b6/src/browser/provider/built-in/dedicated/base.js#L53
I can add it merge it with first param object if you want.

Reply:
My apologies, forgot about this part, let's leave it as it is for now.